app/controllers/news_controller.rb

```diff
@@ -3,10 +3,20 @@ class NewsController < ApplicationController
   load_permissions_and_authorize_resource
 
   def index
-    @news = News.by_date
+    @news = @news.by_date
+    if params[:category].present?
+      @news = category(@news)
+    end
   end
 
   def show
     @news = News.find(params[:id])
   end
+
+  private
+
+  def category(news)
+    @category = Category.find_by_slug(params[:category])
+    news.joins(:categories).where(categories: { slug: params[:category] })
```

Comment:
This is good, the where gets folded into the join. Have you been reading some SQL tutorials? :)